Examples/Sources/AWSXRayInstrumentExample/main.swift

```diff
@@ -0,0 +1,35 @@
+import AWSXRayInstrument
+import AWSXRayRecorder
+import Baggage // BaggageContext
+import Instrumentation // InstrumentationSystem
+import NIOHTTP1 // HTTPHeaders
+import NIOInstrumentation // HTTPHeadersExtractor
+
+// create and boostrap the instrument
+let instrument = XRayRecorder(emitter: XRayLogEmitter(), config: .init(logLevel: .debug))
+InstrumentationSystem.bootstrap(instrument)
+
+let tracer = InstrumentationSystem.tracer // the instrument
+
+// extract the context from HTTP headers
+let headers = HTTPHeaders([
+    ("X-Amzn-Trace-Id", "Root=1-5759e988-bd862e3fe1be46a994272793;Sampled=1"),
+])
+var baggage = BaggageContext()
+tracer.extract(headers, into: &baggage, using: HTTPHeadersExtractor())
+
+// TODO: create instrumented HTTP client
+
+// trace
+var span = tracer.startSpan(named: "Span 1", context: baggage)
+span.setAttribute("Attribute 1", forKey: "key1")
+span.addEvent(.init(name: "Event"))
+
+var span2 = tracer.startSpan(named: "Span 2", context: baggage)
+span.setAttribute("Attribute 2", forKey: "key2")
+
+span.end()
+span2.end()
+
+// (tracer as? XRayRecorder)?.wait()
+instrument.wait()
```

Comment:
@slashmo created a simple example, we do need HTTP client for full demo ;-)

```
2020-07-24T00:44:47+0800 info xray.recorder.fb8fcf5e : Segment de3cbcd15bad637b ended @ 1595522687.695928
2020-07-24T00:44:47+0800 debug xray.recorder.fb8fcf5e : Segment de3cbcd15bad637b parent has not been sent
2020-07-24T00:44:47+0800 info xray.recorder.fb8fcf5e : Segment aaa77a1bb787f81d ended @ 1595522687.696883
2020-07-24T00:44:47+0800 info xray.recorder.fb8fcf5e : Segment aaa77a1bb787f81d emitted @ 1595522687.696967
2020-07-24T00:44:47+0800 info xray.recorder.fb8fcf5e : Segment b16ba007e58ea1ac ended @ 1595522687.696969
2020-07-24T00:44:47+0800 debug xray.recorder.fb8fcf5e : Segment aaa77a1bb787f81d has 0 subsegments inProgress
2020-07-24T00:44:47+0800 info xray.log_emitter.d0ec65f4 : 
{
  "service" : {
    "version" : "aws-xray-sdk-swift"
  },
  "end_time" : 1595522687.696883,
  "subsegments" : [
    {
      "end_time" : 1595522687.6959281,
      "id" : "de3cbcd15bad637b",
      "start_time" : 1595522687.6959109,
      "parent_id" : "aaa77a1bb787f81d",
      "type" : "subsegment",
      "trace_id" : "1-5759e988-bd862e3fe1be46a994272793",
      "name" : "Event"
    }
  ],
  "id" : "aaa77a1bb787f81d",
  "annotations" : {
    "key1" : "Attribute 1",
    "key2" : "Attribute 2"
  },
  "start_time" : 1595522687.695787,
  "trace_id" : "1-5759e988-bd862e3fe1be46a994272793",
  "name" : "Span 1"
}
2020-07-24T00:44:47+0800 info xray.recorder.fb8fcf5e : Segment b16ba007e58ea1ac emitted @ 1595522687.702642
2020-07-24T00:44:47+0800 debug xray.recorder.fb8fcf5e : Segment b16ba007e58ea1ac has 0 subsegments inProgress
2020-07-24T00:44:47+0800 info xray.log_emitter.d0ec65f4 : 
{
  "trace_id" : "1-5759e988-bd862e3fe1be46a994272793",
  "id" : "b16ba007e58ea1ac",
  "service" : {
    "version" : "aws-xray-sdk-swift"
  },
  "name" : "Span 2",
  "start_time" : 1595522687.6968629,
  "end_time" : 1595522687.696969
}
```

Reply:
Cool 😎 Until I finished the `AsyncHTTPClient` integration you could also copy our `InstrumentedHTTPClient` which just wraps the `HTTPClient`: https://github.com/slashmo/gsoc-swift-tracing/blob/main/UseCases/Sources/ManualAsyncHTTPClient/main.swift#L38

Reply:
the problem is it only prints headers ;-)
I made a quick proof of concept implementation of a `BetterHTTPClient` see https://github.com/slashmo/gsoc-swift-tracing/issues/95